src/platform_impl/windows/window_state.rs

```diff
@@ -351,7 +361,17 @@ impl WindowFlags {
     if diff != WindowFlags::empty() {
       let (style, style_ex) = new.to_window_styles();
 
+      let mut close_item_flags = MF_BYCOMMAND;
+      if new.contains(WindowFlags::CLOSABLE) {
+        close_item_flags |= MF_ENABLED;
+      } else {
+        close_item_flags |= MF_DISABLED | MF_GRAYED;
+      }
+
       unsafe {
+        let window_menu = GetSystemMenu(window, false);
+        EnableMenuItem(window_menu, SC_CLOSE, close_item_flags);
+
```

Comment:
```suggestion
    if diff.contains(WindowFlags::CLOSABLE) {
      unsafe {
        let system_menu = GetSystemMenu(window, false);
        EnableMenuItem(
          system_menu,
          SC_CLOSE,
          MF_BYCOMMAND
            | if new.contains(WindowFlags::CLOSABLE) {
              MF_ENABLED
            } else {
              MF_DISABLED | MF_GRAYED
            },
        );
      }
    }
```

And move it above L355

Reply:
I tried this but it doesn't work on window creation (with `WindowBuilder::with_closable(false)`) except by removing the `if diff.contains(WindowFlags::CLOSABLE)`.

Reply:
my bad, that's probably because of line 938 in `windows/window.rs`
it should be removed and instead call `window.set_closable()` later after creating the window at L1041

Reply:
Ah, could be… I can't remember if I tried that or not. I'll try it now.

Reply:
Still doesn't work properly… I'll play with it a bit more in the morning

Reply:
It works if I do this:
```rust
    if self == WindowFlags::empty() || diff.contains(WindowFlags::CLOSABLE) {
```
Would that be okay?

Reply:
I pushed a fix, it needed to have a default so the diffing would work.